src/core/middleware.py

```diff
@@ -0,0 +1,24 @@
+from django.views.decorators.cache import patch_cache_control
+from django.utils.cache import patch_vary_headers
+
+class DownstreamCaching(object):
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        headers = {
+            'public': True,
+
+            # could probably bump these values to 10minutes.
+
+            'max-age': 60 * 5, # 5 minutes, 300 seconds
+            'stale-while-revalidate': 60 * 5, # 5 minutes, 300 seconds
+            'stale-if-error': (60 * 60) * 24, # 1 day, 86400 seconds
+        }
+        response = self.get_response(request)
+
+        if not response.get('Cache-Control'):
+            patch_cache_control(response, **headers)
+        patch_vary_headers(response, ['Accept'])
```

Comment:
IIRC Observer doesn't do content negotiation and is meant to be accessed through browsers? If so, the first point makes it redundant, and second will reduce hit rates dramatically (unless `Accept` headers are normalised at the edge).

Reply:
IIRC Kong is transparent for the `Accept` header, so it won't be normalized. Since browsers access content through links like https://observer.elifesciences.org/report/latest-articles.csv and there is a default for https://observer.elifesciences.org/report/latest-articles (RSS?), this `Vary` value would be redundant?